src/com/google/javascript/jscomp/TypeCheck.java

```diff
@@ -944,6 +944,7 @@ public void visit(NodeTraversal t, Node n, Node parent) {
       case IMPORT_SPEC:
       case IMPORT_SPECS:
       case IMPORT_STAR:
+      case DYNAMIC_IMPORT:
```

Comment:
This isn't right. `import(something)` is typeable.
Its type should have been already set by TypeInference.

DYNAMIC_IMPORT needs its own case that calls a dedicated `visitDynamicImport()` method.
That method should assert that the child node is typed as string and that the `DYNAMIC_IMPORT` node itself has already been given a type by TypeInference.

If you look around at the other `visitBlah()` methods, I'm sure you can find examples of how to do this.

Reply:
Also, need to add test cases to TypeCheckNoTranspileTest and TypeInferenceTest.

Reply:
That helped - I refactored this significantly. I added a check so that the import specifier cannot be undefined or null.